### PR TITLE
[WIP] Update for Fable 1.0 support

### LINKED ISSUE #17: Update for Fable 1.0 support
Update this repo to support Fable 1.0

### PR DESCRIPTION
**Pull request by jgrund**
Fixes #17.

This is WIP, don't merge.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -132,8 +129,6 @@ dependencies {
     compile fileTree(dir: "libs", include: ["*.jar"])
     compile "com.android.support:appcompat-v7:23.0.1"
     compile "com.facebook.react:react-native:+"  // From node_modules
-    compile project(':react-native-image-picker')
```
Comment (forki) [inline android/app/build.gradle]: this is not good we need this for the sample

File: android/app/src/main/java/com/nightwatch/MainApplication.java [hunk 2]
```diff
@@ -25,9 +25,7 @@ protected boolean getUseDeveloperSupport() {
     @Override
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
-            new MainReactPackage(),
```
Comment (forki) [inline android/app/src/main/java/com/nightwatch/MainApplication.java]: same here

File: android/build.gradle [hunk 3]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.0'
+        classpath 'com.android.tools.build:gradle:1.3.1'
```
Comment (forki) [inline android/build.gradle]: please don't downgrade things

**Comment by forki** [comment]
I'm super happy that you try to tackle this. Where can I buy you beer?

File: package.json [hunk 4]
```diff
@@ -3,26 +3,35 @@
...
+    "babel-preset-es2015": "^6.24.0",
+    "fable-elmish": "0.9.0-beta-7",
+    "fable-elmish-react": "0.9.0-beta-4",
+    "fable-loader": "^1.0.0-narumi-905",
+    "fable-powerpack": "1.0.0-narumi-5",
+    "fable-react": "1.0.0-narumi-6",
+    "haul-cli": "0.5.0",
+    "react-native-cli": "^2.0.1",
+    "webpack": "^2.2.1",
+    "webpack-dev-server": "^2.4.2"
```
Comment (et1975) [inline package.json]: probably unused?

File: package.json [hunk 5]
```diff
@@ -3,26 +3,35 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "firstbuild": "npm install && fable",
-    "prebuild": "npm install",
-    "build": "fable --watch"
+    "build": "webpack",
+    "start": "webpack-dev-server",
```
Comment (et1975) [inline package.json]: what can this one do?

File: android/app/build.gradle (hunk 1 above)
Comment (jgrund) [inline android/app/build.gradle]: FYI, these changes were made when I ran `react-native upgrade`.

Without doing that, I could not launch on my mac.

What version of react-native are you using?

File: package.json (hunk 5 above)
Comment (jgrund) [inline package.json]: These were added by the template, I'll remove.

File: package.json (hunk 4 above)
Comment (jgrund) [inline package.json]: These were added by the template, I'll remove.

File: android/app/build.gradle (hunk 1 above)
Comment (et1975) [inline android/app/build.gradle]: Yeah, the native parts are really edit-at-your-own-risk. Not sure if it's a flaw in the tooling or we are not extending it correctly.

Comment (forki) [inline android/app/build.gradle]: Yes I know. That command is stupid and don't understand that changes were made to the files. So you can need to stage carefully. 
That said there is a new replacement called "react-native-upgrafe-git" or something like that. This looks into history and is much better for such situations.

File: package.json (hunk 5 above)
Comment (et1975) [inline package.json]: Ah, Fable.Templates.Elmish.React? That one is for the web, we'll need a new one for the native once you figured out what goes where :)

File: android/app/build.gradle (hunk 1 above)
Comment (jgrund) [inline android/app/build.gradle]: I guess the question I should be asking is, is it possible to run this project on a mac?

**Comment by jgrund** [comment]
Repushing with config changes reverted.

**Comment by forki** [comment]
hi, what's the status? thx

**Comment by jgrund** [comment]
Haven't had much more time to work on this unfortunately. 

Last time I looked, there were some binding updates that needed to be made. 

I'm happy to collaborate on this if someone else wants to carry the torch; Otherwise I'll try to get back to it as soon as I'm able to.

**Comment by alfonsogarciacaro** [comment]
I wonder if it may be easier to use @ncave's fable-splitter for this instead of Webpack+Haul. We still need to add a watch mode (and probably also a CLI) but that shouldn't be difficult.

**Comment by forki** [comment]
is this obsolete with #19?

**Comment by alfonsogarciacaro** [comment]
I guess we can close it but I'm still interested to see if we can use Webpack+Haul with React Native if @jgrund has time to finish it.

**Comment by jgrund** [comment]
I'm going to close this. It's probably worth revisiting once the other PR that introduces 1.0 support lands.

Btw, Facebook recently extracted the react native builder into it's own project https://github.com/facebook/metro-bundler.

We should be looking at that instead of haul going forward I think.

**Pull request closed without merging**